MagneticField/VolumeBasedEngine/src/MagGeometry.cc

```diff
@@ -63,12 +66,15 @@ MagGeometry::MagGeometry(int geomVersion,
   //FIXME assume sectors are already sorted in phi
   //FIXME: PeriodicBinFinderInPhi gets *center* of first bin
   int nEBins = theESectors.size();
-  theEndcapBinFinder = new PeriodicBinFinderInPhi<float>(theESectors.front()->minPhi() + Geom::pi() / nEBins, nEBins);
+  if (nEBins > 0)
+    theEndcapBinFinder = new PeriodicBinFinderInPhi<float>(theESectors.front()->minPhi() + Geom::pi() / nEBins, nEBins);
 }
 
 MagGeometry::~MagGeometry() {
-  delete theBarrelBinFinder;
-  delete theEndcapBinFinder;
+  if (theBarrelBinFinder != nullptr)
```

Comment:
why not ```std::unique_ptr```?

Reply:
std::unique_ptr was not existing in C++ when the code was written. Overhauling the package to use unique_ptr consistently (ie not just here, but in all similar cases, which are quite a few) is obviously possible, but I think that is beyond the scope of the current PR, and personally I don't think it's worth the effort - "if it ain't broke..."

Reply:
I agree with Nicola.